src/Router.sol

```diff
@@ -143,17 +152,14 @@ contract Router is ReentrancyGuard, IRouter, ScheduledUpgradeable, AccessControl
         uint256 amountOut,
         uint256 solverFee,
         uint256 dstChainId,
-        address recipient
+        address recipient,
+        Hook[] calldata preHooks,
```

Comment:
does it cost much to have multiple functions for each config? 

we could provide:
`requestSwapWithHooks()` - a rename of this
`requestStableSwap()` that calls `requestSwapWithHooks` with `amountOut = amountIn`
`requestSwap()` that calls `requestSwapWithHooks` with empty hooks

I noticed at the hacker house this week that any extra parameter or config leads to big confusion